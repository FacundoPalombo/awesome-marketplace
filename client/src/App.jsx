import React from "react";
import styles from "./App.module";
import { Switch, Route } from "react-router-dom";

import { Detail } from "./container/Detail";
import { Search } from "./container/Search";

import { InputSearch } from "./components/InputSearch";
import { NotFound } from "./common/NotFound/NotFound";
import { Results } from "./container/Results";

export function App() {
  return (
    <>
      <main id="main" className={styles.main}>
        <InputSearch />
        <Switch>
          <Route exact path="/">
            <Search />
          </Route>
          <Route exact path="/items">
            <Results />
          </Route>
          <Route path="/items/:id">
            <Detail />
          </Route>
          <Route path="*">
            <NotFound />
          </Route>
        </Switch>
      </main>
    </>
  );
}
