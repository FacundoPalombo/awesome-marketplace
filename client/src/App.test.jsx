import React from "react";

import { App } from "./App";
import { renderWithRouter } from "./utils/tests/renderWithRouter";

describe("GIVEN App", () => {
  describe("WHEN render", () => {
    it("THEN should match to snapshot", () => {
      const { asFragment } = renderWithRouter(<App />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
