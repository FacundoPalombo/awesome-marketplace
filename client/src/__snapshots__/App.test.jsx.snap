// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN App WHEN render THEN should match to snapshot 1`] = `
<DocumentFragment>
  <main
    class="main"
    id="main"
  >
    <form
      class="form"
    >
      <div
        class="box"
      >
        <a
          class="link"
          href="/"
        >
          <img
            alt="browse for items"
            class="brand"
            src="file-mock"
            srcset="file-mock 1x, file-mock 2x"
          />
        </a>
        <input
          class="input"
          id="search_input"
          name="search"
          placeholder="Nunca dejes de buscar"
          type="search"
          value=""
        />
        <button
          class="button"
          type="submit"
        >
          <img
            alt="browse for items"
            class="button"
            src="file-mock"
            srcset="file-mock 1x, file-mock 2x"
          />
        </button>
      </div>
    </form>
    <div>
      search
    </div>
  </main>
</DocumentFragment>
`;
